### PR TITLE
Add miri tests

### PR DESCRIPTION
**Pull request by Nugine**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -64,3 +64,16 @@ jobs:
...
+
+  miri:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@nightly
+        with: 
+          components: miri
+      - run: |
+          cargo miri test
```
Comment (DoumanAsh) [inline .github/workflows/rust.yml]: run tests only with `feature alloc`
there is no point in running without it

**Review by DoumanAsh — COMMENTED** [review]
Can you please rebase?
I fixed `mut ptr -> const ptr -> mut ptr` cast

File: .github/workflows/rust.yml (hunk 1 above)
Comment (Nugine) [reply .github/workflows/rust.yml]: Done

**Review by DoumanAsh — APPROVED** [review]

**Comment by DoumanAsh** [comment]
Thank you

**Pull request merged**